packages/video_player/ios/Classes/VideoPlayerPlugin.m

```diff
@@ -82,14 +83,13 @@ - (void)addObservers:(AVPlayerItem*)item {
             options:NSKeyValueObservingOptionInitial | NSKeyValueObservingOptionNew
             context:playbackBufferFullContext];
 
-  [[NSNotificationCenter defaultCenter] addObserverForName:AVPlayerItemDidPlayToEndTimeNotification
+  _notificationObserverId = [[NSNotificationCenter defaultCenter] addObserverForName:AVPlayerItemDidPlayToEndTimeNotification
```

Comment:
It might be cleaner to just use the `addObserver:self` here since we are just using mainQueue. This way we can separate the observing implementation out of the `addObservers` method.